docs/Guidelines.md

```diff
@@ -0,0 +1,369 @@
+        virtual void ConfigureDefines(Configuration conf, Target target) { ... }
+    }
+    [Generate]
+    class MySubProject : MyProject
+    {
+        override void Configure(Configuration conf, Target target)
+        {
+            base.Configure(conf, target);
+            ...
+        }
+    }
+```
+
+The level of granularity of these virtual methods is up to you.  It can be very small:
+
+```csharp
+        [Configure]
+        void Configure(Configuration conf, Target target)
+        {
+            ...
+            ConfigureStringPooling(conf, target);
+            ...
+        }
+        public virtual void ConfigureStringPooling(Configuration conf, Target target)
+        {
+            conf.Options.Add(Sharpmake.Options.Vc.Compiler.StringPooling.Disable);
+        }
+```
+
+Multiple Configure functions can also be used, and Sharpmake is offering a feature to make them specific a to subset of the targets in the project:
+
+```csharp
+[Configure(Platform.win32 | Platform.win64)]
+void ConfigurePs4(Configuration conf, Target target)
+{
+    ...
+}
+
+[Configure(Platform.ps4)]
+void ConfigurePs4(Configuration conf, Target target)
+{
+    ...
+}
+ 
+[Configure(Platform.durango)]
+void ConfigureDurango(Configuration conf, Target target)
+{
+    ...
+}
+```
+
+If doing multiple Configure functions, it is strongly recommanded to avoid depending on Configure methods execution order.  If you do, the attribute ```[ConfigureOrder]``` can be used.  Sharpmake is using the order of declaration in the class as much as possible, but then changing a Configure method from virtual to override with code moved in a base class can change the execution order.  Many programmers are completely surprised when simply moving code around is changing the produced result.  For these reasons it is suggested that different Configure functions work on different and independent things.
+
+### Dependency System
+
+The dependency system is offering tools to write things about a library a single time, making sure dependent projects get all 
+
+For example, while Sharpmake allows you to do the following:
+
+```csharp
+conf.LibraryFiles.Add("Iphlpapi.lib");
+```
+
+It might be better to take the time to define everything for that library once:
+
+```csharp
+[Sharpmake.Export]
+class IPHelperAPI
+{
+    void Configure(Configuration conf, Target target)
+    {
+        ...  // configure for target .lib, needed include path and define for dependents
+    }
+}
+```
+
+That project can then be added as a dependency.
+
+### Public static functions
+
+C# public static functions can offer an easy way to write code a single time.  Suppose you have the following:
+
+```csharp
+class MyBaseProject : Project
+{
+    public MyBaseProject : base(typeof(Target))
+    {
+        AddTargets(new Target(
+            Platform.win32 | Platform.win64, 
+            DevEnv.vs2015,
+            ...));
+    }
+}
+```
+
```

Comment:
Maybe should be also talking of the way we now add dependencies for C# dlls using enums for external dependencies.

Reply:
Since 100% .sharpmake.cs, will mention that in a separate .md.